tests/firedrake/regression/test_adjoint_operators.py

```diff
@@ -112,8 +112,8 @@ def test_interpolate_scalar_valued(rg):
 def test_interpolate_vector_valued():
     mesh = UnitSquareMesh(10, 10)
     V1 = VectorFunctionSpace(mesh, "CG", 1)
-    V2 = VectorFunctionSpace(mesh, "DG", 0)
-    V3 = VectorFunctionSpace(mesh, "CG", 2)
```

Comment:
@dham and I discussed this test, which interpolates from V2=DG0 into V3=CG2. The interpolation is not well defined and the test passes with a matrix-explicit adjoint, but fails with the matfree adjoint. The conclusion was that however attempts such ill-defined operator derseve whatever comes to them.

The whole point of this test is to test the adjoint of vector-valued interpolation, so I just used the vector variant of the spaces used in the scalar test above.